### PR TITLE
DENG-392 Add online/offline attribution to the `custom-data` URL param

### PR DESCRIPTION
**Pull request by quiiver**
https://mozilla-hub.atlassian.net/browse/DENG-392

### CODE REVIEW

File: ingestion-core/src/main/java/com/mozilla/telemetry/ingestion/core/Constant.java [hunk 1]
```diff
@@ -29,6 +29,7 @@ private Attribute() {
     public static final String GEO_SUBDIVISION1 = "geo_subdivision1";
     public static final String GEO_SUBDIVISION2 = "geo_subdivision2";
     public static final String GEO_DMA_CODE = "geo_dma_code";
+    public static final String IMPROVE_SUGGEST_EXPERIENCE = "improve_suggest_experience_check";
```
Comment (ncloudioj) [inline ingestion-core/src/main/java/com/mozilla/telemetry/ingestion/core/Constant.java:32]: Per [schema](https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/main/templates/contextual-services/quicksuggest-impression/quicksuggest-impression.1.schema.json#L66), this should be `improve_suggest_experience_checked`.

My fault, there was a typo in the ticket description.

**Review by ncloudioj — APPROVED** [review]
LGTM. r=nanj

File: ingestion-beam/src/test/java/com/mozilla/telemetry/contextualservices/ParseReportingUrlTest.java [hunk 2]
```diff
@@ -244,6 +244,76 @@ public void testDmaCode() {
...
+        new PubsubMessage(
+            Json.asBytes(clickPayload.put(Attribute.IMPROVE_SUGGEST_EXPERIENCE, "true")),
+            attributes),
+        new PubsubMessage(Json.asBytes(clickPayload), attributes));
+
+    Result<PCollection<SponsoredInteraction>, PubsubMessage> result = pipeline //
+        .apply(Create.of(input)) //
+        .apply(ParseReportingUrl.of(URL_ALLOW_LIST));
+
+    // We expect 3 successful messages.
```
Comment (relud) [inline ingestion-beam/src/test/java/com/mozilla/telemetry/contextualservices/ParseReportingUrlTest.java:280]:
```suggestion
    // We expect 4 successful messages.
```

File: ingestion-beam/src/test/java/com/mozilla/telemetry/contextualservices/ParseReportingUrlTest.java [hunk 3]
```diff
@@ -244,6 +244,76 @@ public void testDmaCode() {
...
+              }
+            } else if (doctype.equals("quicksuggest-click")) {
+              if (interaction.getScenario().equals(SponsoredInteraction.ONLINE)) {
+                Assert.assertTrue(reportingUrl.contains(
+                    String.format("%s=%s", ParsedReportingUrl.PARAM_CUSTOM_DATA, "1_online")));
+              } else {
+                Assert.assertTrue(reportingUrl
+                    .contains(String.format("%s=%s", ParsedReportingUrl.PARAM_CUSTOM_DATA, "1")));
+              }
+            }
```
Comment (relud) [inline ingestion-beam/src/test/java/com/mozilla/telemetry/contextualservices/ParseReportingUrlTest.java:305]: this checks that `setScenario`, but to test that  `parseScenario` works it needs to assert that every branch is taken a specific number of times

**Review by relud — APPROVED** [review]
r+wc

File: ingestion-beam/src/test/java/com/mozilla/telemetry/contextualservices/ParseReportingUrlTest.java (hunk 3 above)
Comment (quiiver) [reply ingestion-beam/src/test/java/com/mozilla/telemetry/contextualservices/ParseReportingUrlTest.java]: Thanks, I had meant to go back and add something like that but forgot before I created the review.

**Pull request merged**